buildlog/buildlog.go

```diff
@@ -0,0 +1,102 @@
+package buildlog
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+)
+
+var readInterval time.Duration = 10 * time.Millisecond
+
+// Logger outputs logs
+type Logger interface {
+	Run() chan error
+	Stop()
+}
+
+type log struct {
+	ctx    context.Context
+	file   io.Reader
+	writer io.Writer
+	cancel context.CancelFunc
+}
+
+type logLine struct {
+	Time     int64  `json:"t"`
+	Message  string `json:"m"`
+	Line     int    `json:"n"`
+	StepName string `json:"s"`
+}
+
+// New creates new Logger interface.
+func New(ctx context.Context, filepath string, writer io.Writer) (Logger, error) {
+	log := log{
+		writer: writer,
+	}
+
+	var err error
+	log.file, err = os.OpenFile(filepath, os.O_RDONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		return log, fmt.Errorf("failed to open raw build log file :%w", err)
+	}
+
+	log.ctx, log.cancel = context.WithCancel(ctx)
+
+	return log, nil
+}
+
+func (l log) Run() chan error {
+	errChan := make(chan error)
+	go l.run(errChan)
+	return errChan
+}
+
+func (l log) Stop() {
+	l.cancel()
+	return
+}
+
+func (l log) run(errChan chan error) {
```

Comment:
`chan error` does not need.
I think it is ok to continue the build if the log parsing fails.

Reply:
I fixed it.